### PR TITLE
USB host stack!

### PR DESCRIPTION
**Pull request by kuldeepdhaka**

### CODE REVIEW

**Comment by ChuckM** [comment]
Hi Kuldeep, I pulled this into mainline and tried to do a build and it complained about for loop initializations. I'll note the line in the files.

File: lib/usbh/usbh_dev_enum.c [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+/**
+ * Allocate the next device address from the host
+ * @param host USB Host
+ * @return valid non-zero 7bit address
+ * @return zero on failure
+ */
+static uint8_t alloc_device_address(usbh_host *host)
+{
+	for (unsigned i = 0; i < 127; i++) {
```
Comment (ChuckM) [inline lib/usbh/usbh_dev_enum.c]: Declaring the variable inside the for loop here, gets a complaint from the compiler.

File: lib/usbh/usbh_dev_enum.c [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+		/* fail after 127 attempts! (address space is full) */
+
+		uint8_t addr = host->next_device_address & 0x7F;
+		addr = (!addr) ? 1 : addr;
+		host->next_device_address = (addr == 0x7F) ? 1 : (addr + 1);
+
+		for (unsigned j = 0; j < DEVICE_ARRAY_LENGTH; j++) {
```
Comment (ChuckM) [inline lib/usbh/usbh_dev_enum.c]: And here.

**Comment by kuldeepdhaka** [comment]
im abandoning (and closing) the PR - i dont have energy to manage it.
If someone is looking for usb host stack, please take the effort to do copy the code from unicore-mx into libopencm3 or use unicore-mx.
(usb host has been merged in unicore-mx and this_pr,unicore-mx code has diverged)

**Pull request closed without merging**